Week4/week 4.ipynb

```diff
@@ -0,0 +1,335 @@
+    "                                after delete \\\n",
+    "                                on contact\\\n",
+    "                                begin\\\n",
+    "                                insert into logtable\\\n",
+    "                                values(old.fname,old.lname,old.contact,datetime('now','localtime'),'DELETE');\\\n",
+    "                                end;\"\n",
+    "\n",
+    "    curobj.execute(log_table_delete_trigger)\n",
+    "    log_table_update_trigger=\"create trigger IF NOT EXISTS update_trigger\\\n",
+    "                                after update\\\n",
+    "                                on contact\\\n",
+    "                                begin\\\n",
+    "                                insert into logtable\\\n",
+    "                                values(new.fname,new.lname,new.contact,datetime('now','localtime'),'UPDATE(After)');\\\n",
+    "                                insert into logtable\\\n",
+    "                                values(old.fname,old.lname,old.contact,datetime('now','localtime'),'UPDATE(Before)');\\\n",
+    "                                end;\"\n",
+    "\n",
+    "    curobj.execute(log_table_update_trigger)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 23,
+   "id": "14156ea3",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def menu():\n",
+    "    print('Enter 1 To Insert Contact No')\n",
+    "    print('Enter 2 To Delete Contact No')\n",
+    "    print('Enter 3 To Update Contact No')\n",
+    "    print('Enter 4 To Search Contact No')\n",
+    "    print('Enter 0 To Quit')\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 24,
+   "id": "27eae152",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def insert_contact():\n",
+    "    try:\n",
+    "            query = \"insert into contact \\\n",
+    "                    values(?,?,?,?,?)\"\n",
+    "            \n",
+    "            First_Name=input('Enter Your First Name :')\n",
+    "            Last_Name=input('Enter Your Last Name :')\n",
+    "            Contact = int(input('Enter Your Contact No :'))\n",
+    "            Email=input('Enter Your Email ID :')\n",
+    "            City=input('Enter Your City :')\n",
+    "            main_list=[First_Name,Last_Name,Contact,Email,City]\n",
+    "            curobj.execute(query,main_list)\n",
+    "            print()\n",
+    "            print(\"Record(s) are/is Inserted Successfully\")\n",
+    "            print()\n",
+    "    except Error as e:\n",
+    "            print()\n",
+    "            print(e)\n",
+    "            print()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 25,
+   "id": "0f25e53c",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def delete_contact():\n",
+    "    try:\n",
+    "            Name = input('Enter Your Contact First Name That You Want To Delete : ')\n",
+    "            query = f\"delete from contact\\\n",
```

Comment:
is there any other way to format this line?

Reply:
I am not sure but I know one other way that is by directly writing in execute section like cur.execute(f“delete from contact where fname={Name}”)